### DISCUSSION TITLE
Is there a way to disable pages altogether?

### DISCUSSION THREAD

**Discussion by jakiestfu** [nextauthjs/next-auth#2330] [Help]
### Question 💬

I've built my own UI for handling sessions altogether, so I'd like to disable the automatically generated pages.

I've tried adding the following to no avail. I'm using the email and github providers.

```
...
   pages: {
      signIn: null,
      signOut: null,
      error: null,
      verifyRequest: null,
      newUser: null,
    },
...
```

### How to reproduce ☕️

Unable to reproduce my desired behavior.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by Pruxis** [comment]
I would like to be able to disable these pages too.
I've already checked the source code and this seems to be pretty easy to add.

However what would be the desired approach for this @iaincollins ?

If we could discuss this then I'd be happy to open a PR to add this functionality, as we really dislike "non-owned" pages to be accessible to the end-user.

**Reply by balazsorban44** [reply]
Just configure all of them to point to `/` or to any of your custom pages.

You can see here what is expected to be handled on these pages https://next-auth.js.org/configuration/pages

**Reply by iaincollins** [reply]
^ What @balazsorban44 said.

An option to simply setting `pages: false` to map all responses to `/` as shorthand seems not unreasonable to me.

**Reply by balazsorban44** [reply]
https://github.com/nextauthjs/next-auth/issues/1606#issuecomment-808955181

**Reply by phelanma** [reply]
(following the advise given here*) 
Setting `/` for pages will result '?callbackUrl=http...' being appended to url; which is not really the desire? should it not just redirect without callback being appended?

**Reply by pingustar** [reply]
@phelanma did you find a solution to removing the appended callbackUrl query param?

**Reply by zekepari** [reply]
@pingustar Did you find a solution for this?

**Reply by chev0004** [reply]
@pingustar have you found a solution yet?

**Reply by fxbayuanggara** [reply]
You can add the provider id to signIn method, for example, `signIn('keycloak')`

**Comment by alihesari** [comment]
Did anyone find a solution to this?

I use Keycloak as a provider. Keycloak has a signin page but sometimes nextauthjs redirects users to this page. I want to disable this page and not redirect users to this page. The route of this page is `api/auth/signin?csrf=true`


<img width="901" alt="image" src="https://github.com/nextauthjs/next-auth/assets/8302633/1c94dc53-e40d-43be-91ba-823118cef0f7">

**Comment by reslear** [comment]
any news? I would like to use only the api

**Reply by ericmesmo** [reply]
Did you find a way?

**Reply by reslear** [reply]
I decided not to migrate. I'm using passportjs.

**Comment by kr-yeon** [comment]
+1

**Comment by svarup** [comment]
+1

**Comment by mayur-padshala** [comment]
I was able to prevent the pages by using nextjs middleware

```javascript
if (req.method === "GET") {
  if (req.nextUrl.pathname === "/api/auth/signin") {
    return NextResponse.redirect(req.nextUrl.origin, 308);
  }
  if (req.nextUrl.pathname === "/api/auth/signout") {
    return NextResponse.redirect(req.nextUrl.origin, 308);
  }
}
```

**Comment by byayex** [comment]
+1

**Comment by svarup** [comment]
We can set [pages](https://authjs.dev/getting-started/session-management/custom-pages).

```
Pages: {
      signIn: '/login',
      signOut: '/logout',
      error: '/error',
      verifyRequest: '/login',
},
 providers: [...]
 ```
